spec/javascripts/components/accordion-spec.js

```diff
@@ -49,22 +63,50 @@ describe('Accordion component', function () {
 
   afterEach(function () {
     document.body.removeChild(container)
+    if (GOVUK.analytics.trackEvent.calls) {
+      GOVUK.analytics.trackEvent.calls.reset()
+    }
+  })
+
+  it('applies custom attributes to click event if specified in section', function () {
+    startAccordion()
+    document.querySelector('.govuk-accordion__section-header button').click()
+    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('pageElementInteraction', 'accordionOpened', { transport: 'beacon', label: 'Accordion Section Heading Text', dimension114: 1 })
+  })
+
+  it('applies custom attributes to click event if specified in show/hide section', function () {
+    var object = {
+      'track-options': {
+        dimensionExample: 'Example value'
+      }
+    }
+
+    accordion.setAttribute('data-show-all-attributes', JSON.stringify(object))
+
+    startAccordion()
+    document.querySelector('.gem-c-accordion__show-all').click()
+    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('pageElementInteraction', 'accordionClosed', Object({ transport: 'beacon', label: 'Hide all sections', dimensionExample: 'Example value' }))
   })
 
   it('applies given data attributes to the show/hide all link', function () {
     var object = {
       ui: {
         type: 'type value',
         section: 'section value'
+      },
+      'track-options': {
+        dimensionExample: 'Example value'
       }
     }
+
     var wrappingObject = {
       'gtm-event-name': 'example',
-      'show-all-attributes': JSON.stringify(object)
+      ...object
```

Comment:
Is there a particular reason for using spread syntax here, over the previous method? I ask because it's not supported in IE and it would be good to be able to run the tests in IE if we need to, and this would block that.